dynamodb-enhanced/deployment/src/main/java/io/quarkus/amazon/dynamodb/enhanced/runtime/EnhancedProcessor.java

```diff
@@ -0,0 +1,200 @@
+import io.quarkus.deployment.builditem.LaunchModeBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.gizmo.Gizmo;
+import io.quarkus.runtime.LaunchMode;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbBean;
+
+public class EnhancedProcessor {
+    private static final String FEATURE = "amazon-dynamodb-enhanced";
+    public static final String CLASS_NAME_BEAN_TABLE_SCHEMA = "software.amazon.awssdk.enhanced.dynamodb.mapper.BeanTableSchema";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    AdditionalBeanBuildItem producer() {
+        return AdditionalBeanBuildItem.unremovableOf(DynamodbEnhancedClientProducer.class);
+    }
+
+    @BuildStep
+    public void registerClassesForReflectiveAccess(
+            CombinedIndexBuildItem combinedIndexBuildItem,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
+        for (AnnotationInstance i : combinedIndexBuildItem
+                .getIndex()
+                .getAnnotations(DotName.createSimple(DynamoDbBean.class.getName()))) {
+            ClassInfo classInfo = i.target().asClass();
+            reflectiveClass.produce(
+                    new ReflectiveClassBuildItem(true, false, classInfo.name().toString()));
+        }
+
+        // When we add DefaultAttributeConverterProvider and BeanTableSchemaAttributeTags here, we do not need the DynamodbEnhancedFeature in runtime module.
+        //reflectiveClass.produce(
+        //        new ReflectiveClassBuildItem(true, false,
+        //                "software.amazon.awssdk.enhanced.dynamodb.DefaultAttributeConverterProvider"));
+        //reflectiveClass.produce(
+        //        new ReflectiveClassBuildItem(true, false,
+        //                "software.amazon.awssdk.enhanced.dynamodb.internal.mapper.BeanTableSchemaAttributeTags"));
+
+    }
+
+    @BuildStep
+    void maybeApplyClassTransformation(
+            BuildTimeConfig buildTimeConfig,
+            BuildProducer<BytecodeTransformerBuildItem> transformers,
+            LaunchModeBuildItem launchModeBuildItem) {
+        if (shouldApplyClassTransformation(buildTimeConfig, launchModeBuildItem)) {
+            applyClassTransformation(transformers);
+        }
+    }
+
+    private void applyClassTransformation(BuildProducer<BytecodeTransformerBuildItem> transformers) {
+        transformers.produce(
+                new BytecodeTransformerBuildItem(
+                        CLASS_NAME_BEAN_TABLE_SCHEMA, new MethodCallRedirectionVisitor()));
+    }
+
+    /**
+     * Decides if the transformation should be applied. Predominantly this should only happen in
+     * non-production builds because only a single {@link ClassLoader} is used there making this "fix"
+     * useless and probably worsens performance. Second, we do not apply the transformation when
+     * requested through config.
+     */
```

Comment:
Could you explain what you're trying to do here with this transformation? I had a look at your discussions but couldn't understand what's going on at a quick glance.
Moreover we need a comment that actually explains the rationale of this change. Here you explain why it's not useful in all cases but you have no idea what the purpose is.